### PR TITLE
DRILL-6575: Add store.hive.conf.properties option to allow set Hive properties at session level

### PR DESCRIPTION
**Pull request by arina-ielchiieva**
Details in [DRILL-6575](https://issues.apache.org/jira/browse/DRILL-6575).

### CODE REVIEW

**Comment by arina-ielchiieva** [comment]
@vdiravka could you please review.

File: exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/SqlConverter.java [hunk 1]
```diff
@@ -187,12 +187,11 @@ public SqlNode parse(String sql) {
...
   public SqlNode validate(final SqlNode parsedNode) {
     try {
-      SqlNode validatedNode = validator.validate(parsedNode);
-      return validatedNode;
+      return validator.validate(parsedNode);
     } catch (RuntimeException e) {
       UserException.Builder builder = UserException
           .validationError(e)
-          .addContext("SQL Query", sql);
+          .addContext("SQL Query", (String) null);
```
Comment (vdiravka) [inline exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/SqlConverter.java]: Why do we need to use it? `addContext()` with one parameter exists

File: exec/java-exec/src/main/java/org/apache/drill/exec/opt/BasicOptimizer.java [hunk 2]
```diff
@@ -241,7 +233,7 @@ public PhysicalOperator visitFilter(final Filter filter, final Object obj) throw
     }
 
     @Override
-    public PhysicalOperator visitUnnest(final Unnest unnest, final Object obj) throws OptimizerException {
+    public PhysicalOperator visitUnnest(final Unnest unnest, final Object obj) {
       return new org.apache.drill.exec.physical.config.UnnestPOP(null, unnest.getColumn());
```
Comment (vdiravka) [inline exec/java-exec/src/main/java/org/apache/drill/exec/opt/BasicOptimizer.java]: add `org.apache.drill.exec.physical.config.` to import statements

File: contrib/storage-hive/core/src/main/java/org/apache/drill/exec/store/hive/HiveUtilities.java [hunk 3]
```diff
@@ -471,20 +498,28 @@ public static void restoreColumns(HiveTableWithColumnCache table, HivePartition
...
+  /**
+   * Generates unsupported types exception message with list of supported types
+   * and throws user exception.
+   *
+   * @param unsupportedType unsupported type
+   */
   public static void throwUnsupportedHiveDataTypeError(String unsupportedType) {
     StringBuilder errMsg = new StringBuilder();
-    errMsg.append(String.format("Unsupported Hive data type %s. ", unsupportedType));
+    errMsg.append("Unsupported Hive data type ").append(unsupportedType).append(". ");
```
Comment (vdiravka) [inline contrib/storage-hive/core/src/main/java/org/apache/drill/exec/store/hive/HiveUtilities.java]: `System.getProperty("line.separator")` -> `System.lineSeparator()`
Also change this part of building string without reusing `errMsg` variable.

File: contrib/storage-hive/core/src/main/java/org/apache/drill/exec/store/hive/HiveUtilities.java [hunk 4]
```diff
@@ -442,7 +469,7 @@ public static Properties getPartitionMetadata(final HivePartition partition, fin
     // Include Table properties in final list in order to not to break SerDes that depend on
     // Table properties. For example AvroSerDe gets the schema from properties (passed as second argument)
     for (Map.Entry<String, String> entry : table.getParameters().entrySet()) {
-      if (entry.getKey() != null && entry.getKey() != null) {
+      if (entry.getKey() != null && entry.getValue() != null) {
```
Comment (vdiravka) [inline contrib/storage-hive/core/src/main/java/org/apache/drill/exec/store/hive/HiveUtilities.java]: replace with stream API chain

File: contrib/storage-hive/core/src/main/java/org/apache/drill/exec/store/hive/HiveUtilities.java [hunk 5]
```diff
@@ -681,5 +716,35 @@ public static boolean containsUnsupportedDataTypes(final Table hiveTable) {
...
+   */
+  public static HiveConf generateHiveConf(Map<String, String> properties) {
+    logger.trace("Override HiveConf with the following properties {}", properties);
+    HiveConf hiveConf = new HiveConf();
+    properties.forEach(hiveConf::set);
+    return hiveConf;
+  }
+
+  /**
+   * Creates HiveConf based on properties in give HiveConf and configuration properties.
```
Comment (vdiravka) [inline contrib/storage-hive/core/src/main/java/org/apache/drill/exec/store/hive/HiveUtilities.java]: given

File: exec/java-exec/src/main/java/org/apache/drill/exec/opt/BasicOptimizer.java [hunk 6]
```diff
@@ -180,16 +172,16 @@ public PhysicalOperator visitLimit(final org.apache.drill.common.logical.data.Li
     @Override
     public PhysicalOperator visitJoin(final Join join, final Object value) throws OptimizerException {
       PhysicalOperator leftOp = join.getLeft().accept(this, value);
-      final List<Ordering> leftOrderDefs = Lists.newArrayList();
-      for(JoinCondition jc : join.getConditions()){
+      final List<Ordering> leftOrderDefs = new ArrayList<>();
+      for(JoinCondition jc : join.getConditions()) {
```
[the remark this answers is not in the document]
Comment (arina-ielchiieva) [reply exec/java-exec/src/main/java/org/apache/drill/exec/opt/BasicOptimizer.java]: Done.

File: exec/java-exec/src/main/java/org/apache/drill/exec/opt/BasicOptimizer.java [hunk 7]
```diff
@@ -180,16 +172,16 @@ public PhysicalOperator visitLimit(final org.apache.drill.common.logical.data.Li
...
         leftOrderDefs.add(new Ordering(Direction.ASCENDING, jc.getLeft()));
       }
       leftOp = new Sort(leftOp, leftOrderDefs, false);
       leftOp = new SelectionVectorRemover(leftOp);
 
       PhysicalOperator rightOp = join.getRight().accept(this, value);
-      final List<Ordering> rightOrderDefs = Lists.newArrayList();
-      for(JoinCondition jc : join.getConditions()){
+      final List<Ordering> rightOrderDefs = new ArrayList<>();
+      for(JoinCondition jc : join.getConditions()) {
```
[the remark this answers is not in the document]
Comment (arina-ielchiieva) [reply exec/java-exec/src/main/java/org/apache/drill/exec/opt/BasicOptimizer.java]: Done.

File: exec/java-exec/src/main/java/org/apache/drill/exec/opt/BasicOptimizer.java (hunk 2 above)
Comment (arina-ielchiieva) [reply exec/java-exec/src/main/java/org/apache/drill/exec/opt/BasicOptimizer.java]: Done.

File: exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/SqlConverter.java (hunk 1 above)
Comment (arina-ielchiieva) [reply exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/SqlConverter.java]: Removed `addContext` since it does not make sense in this case.

**Comment by arina-ielchiieva** [comment]
@vdiravka thanks for the code review. Addressed code review comments.

**Pull request closed without merging**